packages/core/src/decorators/UseRequestContext.ts

```diff
@@ -1,15 +1,15 @@
 import { MikroORM } from '../MikroORM';
 import { RequestContext } from '../utils/RequestContext';
 
-export function UseRequestContext(getContext?: MikroORM | (() => MikroORM)) {
+export function UseRequestContext<T>(getContext?: MikroORM | ((type: T) => MikroORM)): MethodDecorator {
```

Comment:
this needs to be optional, the use case was doing `() => orm` and that needs to be working too
```suggestion
export function UseRequestContext<T>(getContext?: MikroORM | ((type?: T) => MikroORM)): MethodDecorator {
```